cmd/antrea-agent/util_test.go

```diff
@@ -0,0 +1,115 @@
+			expectedipv6:                []net.IP{net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1}, net.IP{0x24, 0x9, 0x40, 0x71, 0x4d, 0x1e, 0x91, 0x43, 0x18, 0x48, 0xec, 0xb6, 0xbf, 0x56, 0xf, 0x8}, net.IP{0x24, 0x9, 0x40, 0x71, 0x4d, 0x1e, 0x91, 0x43, 0x71, 0x24, 0xa0, 0x20, 0x5c, 0x3e, 0x49, 0xa9}},
+		},
+		{
+			name:                        "good path with nodePortAddressesFromConfig",
+			nodePortAddressesFromConfig: []string{"1.2.3.0/24", "1.2.3.4/32"},
+			excludeDevices:              []string{"antrea-egress0", "antrea-ingress0", "kube-ipvs0", "antrea-gw0"},
+			expectedipv4:                []net.IP(nil),
+			expectedipv6:                []net.IP(nil),
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			gotipv4, gotipv6, err := getAvailableNodePortAddresses(tc.nodePortAddressesFromConfig, tc.excludeDevices)
+			assert.NoError(t, err)
+			assert.Equal(t, tc.expectedipv4, gotipv4)
+			assert.Equal(t, tc.expectedipv6, gotipv6)
+		})
+	}
+}
+
+func TestParsePortRange(t *testing.T) {
+	testCases := []struct {
+		name          string
+		portRangeStr  string
+		expectedStart int
+		expectedEnd   int
+		expectedErr   string
+	}{
+		{
+			name:         "wrong port range format",
+			portRangeStr: "200 300",
+			expectedErr:  "wrong port range format: 200 300",
+		},
+		{
+			name:         "length of port range string not equals to 2",
+			portRangeStr: "6110-7000-8000",
+			expectedErr:  "wrong port range format: 6110-7000-8000",
+		},
+		{
+			name:         "wrong port range type value for start",
+			portRangeStr: "wrong-6200",
+			expectedErr:  "strconv.Atoi: parsing \"wrong\": invalid syntax",
+		},
+		{
+			name:         "wrong port range type value for end",
+			portRangeStr: "6100-wrong",
+			expectedErr:  "strconv.Atoi: parsing \"wrong\": invalid syntax",
+		},
+		{
+			name:          "valid range",
+			portRangeStr:  "61000-62000",
+			expectedStart: 61000,
+			expectedEnd:   62000,
+		},
+		{
+			name:         "port range's end smaller than port range's start",
+			portRangeStr: "6000-5000",
+			expectedErr:  "start port must be smaller than end port: 6000-5000",
+		},
+		{
+			name:         "port range's start and port range's end with equal values",
+			portRangeStr: "6100-6100",
+			expectedErr:  "start port must be smaller than end port: 6100-6100",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			start, end, err := parsePortRange(tc.portRangeStr)
+			if tc.expectedErr != "" {
+				assert.EqualError(t, err, tc.expectedErr)
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, tc.expectedStart, start)
+			assert.Equal(t, tc.expectedEnd, end)
```

Comment:
nit: check the whole content of https://github.com/antrea-io/antrea/pull/4349#discussion_r1021071974